flake.nix

```diff
@@ -19,76 +16,70 @@
 
     perSystem = { self', pkgs, lib, ... }:
       let
-        # Generate the typst package for the given nixpkgs instance.
-        packageFor = pkgs:
-          let
-            inherit (lib)
-              importTOML
-              optionals
-              sourceByRegex
-              ;
-            Cargo-toml = importTOML ./Cargo.toml;
-
-            pname = "typst";
-            version = Cargo-toml.workspace.package.version;
-
-            # Crane-based Nix flake configuration.
-            # Based on https://github.com/ipetkov/crane/blob/master/examples/trunk-workspace/flake.nix
-            craneLib = crane.mkLib pkgs;
-
-            # Typst files to include in the derivation.
-            # Here we include Rust files, docs and tests.
-            src = sourceByRegex ./. [
-              "(docs|crates|tests)(/.*)?"
-              ''Cargo\.(toml|lock)''
-              ''build\.rs''
-            ];
-
-            # Typst derivation's args, used within crane's derivation generation
-            # functions.
-            commonCraneArgs = {
-              inherit src pname version;
-
-              buildInputs = (optionals pkgs.stdenv.isDarwin [
-                pkgs.darwin.apple_sdk.frameworks.CoreServices
-                pkgs.libiconv
-              ]) ++ [
-                pkgs.openssl
-              ];
-
-              nativeBuildInputs = [
-                pkgs.installShellFiles
-                pkgs.pkg-config
-                pkgs.openssl.dev
-              ];
-            };
-
-            # Derivation with just the dependencies, so we don't have to keep
-            # re-building them.
-            cargoArtifacts = craneLib.buildDepsOnly commonCraneArgs;
-          in
-          craneLib.buildPackage (commonCraneArgs // {
-            inherit cargoArtifacts;
+        cargoToml = lib.importTOML ./Cargo.toml;
+
+        pname = "typst";
+        version = cargoToml.workspace.package.version;
+
+        # Crane-based Nix flake configuration.
+        # Based on https://github.com/ipetkov/crane/blob/master/examples/trunk-workspace/flake.nix
+        craneLib = crane.mkLib pkgs;
+
+        # Typst files to include in the derivation.
+        # Here we include Rust files, docs and tests.
+        src = lib.sourceByRegex ./. [
+          "(docs|crates|tests)(/.*)?"
+          ''Cargo\.(toml|lock)''
+          ''build\.rs''
+          "rustfmt.toml"
+        ];
```

Comment:
This seems like a good opportunity to move away from `sourceByRegex`, last time I checked it had some problems on MacOS (due to https://github.com/NixOS/nix/issues/1537). Maybe we can just copy and paste what I did for my project, Glistix:
https://github.com/Glistix/glistix/blob/3b578d459db8cb528e0c19682dabe07094626373/nix/glistix.nix#L12-L35